plugins/azure/virtualmachinescaleset/vmssManagedIdentityEnabled.spec.js

```diff
@@ -0,0 +1,82 @@
+        'type': 'Microsoft.Compute/virtualMachineScaleSets',
+        "identity": {
+            "type": "SystemAssigned",
+            "principalId": "5db3ed52-909e-4016-a31f-d2fe043952a4",
+            "tenantId": "d207c7bd-fcb1-4dd3-855a-cfd2f9b651e8"
+        },
+    },
+    {
+        'name': 'test-vmss',
+        'id': '/subscriptions/123/resourceGroups/AQUA-RESOURCE-GROUP/providers/Microsoft.Compute/virtualMachineScaleSets/test-vmss',
+        'type': 'Microsoft.Compute/virtualMachineScaleSets',
+    }
+];
+
+const createCache = (virtualMachineScaleSets) => {
+    let machine = {};
+    if (virtualMachineScaleSets) {
+        machine['data'] = virtualMachineScaleSets;
+    }
+    return {
+        virtualMachineScaleSets: {
+            listAll: {
+                'eastus': machine
+            }
+        }
+    };
+};
+
+describe('vmssManagedIdentityEnabled', function() {
+    describe('run', function() {
+        it('should give passing result if no virtual machine scale sets', function(done) {
+            const cache = createCache([]);
+            vmssManagedIdentityEnabled.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('No existing Virtual Machine Scale Sets found');
+                expect(results[0].region).to.equal('eastus');
+                done();
+            });
+        });
+
+        it('should give unknown result if unable to query for virtual machine scale sets', function(done) {
+            const cache = createCache();
+            vmssManagedIdentityEnabled.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(3);
+                expect(results[0].message).to.include('Unable to query for Virtual Machine Scale Sets:');
+                expect(results[0].region).to.equal('eastus');
+                done();
+            });
+        });
+
+        it('should give passing result if VM scale set has managed identity enabled', function(done) {
+            const cache = createCache([virtualMachineScaleSets[0]]);
+            vmssManagedIdentityEnabled.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('VM scale set has managed identity enabled');
+                expect(results[0].region).to.equal('eastus');
+                done();
+            });
+        });
+
+        it('should give failing result if VM scale set does not have managed identity enabled', function(done) {
+            const cache = createCache([virtualMachineScaleSets[1]]);
+            vmssManagedIdentityEnabled.run(cache, { vmss_approved_extensions: 'healthRepairExtension' }, (err, results) => {
```

Comment:
is this setting? { vmss_approved_extensions: 'healthRepairExtension' }

Reply:
yes

Reply:
but we don't have setting in this plugin

Reply:
my bad resolved